packages/payload/src/fields/baseFields/baseAllLocaleStatusField.ts

```diff
@@ -0,0 +1,38 @@
+import type { Field } from '../config/types.js'
+
+export const baseAllLocaleStatusField: Field[] = [
+  {
+    name: '_allLocaleStatus',
+    type: 'json',
+    admin: {
+      components: {
+        Cell: '@payloadcms/ui/rsc#AllLocaleStatusCell',
+      },
+      hidden: true,
+    },
+    hooks: {
+      afterRead: [
```

Comment:
For performance reasons, I was going to suggest this be an `afterChange` instead, but I guess this wouldn't be as "real-time" as any unsaved documents _after_ this feature is released would not be populated.

This means that this query needs to be as performant as possible. Some easy wins here are specifying `depth` and selecting only the field(s) we need to work with.